packages/ui/src/components/SelectInput/Dropdown.tsx

```diff
@@ -723,6 +723,29 @@ export const Dropdown = ({
   )
   const modalContext = useContext(ModalContext)
 
+  const handleClickOutside = useCallback(
+    (event: MouseEvent) => {
+      if (
+        event.target instanceof Node &&
+        ref.current &&
+        !ref.current.contains(event.target) &&
+        refSelect.current &&
+        !refSelect.current.contains(event.target)
+      ) {
+        setIsDropdownVisible(false)
+      }
+    },
+    [setIsDropdownVisible, refSelect],
+  )
+
+  useEffect(() => {
+    document.addEventListener('mouseup', handleClickOutside)
+
+    return () => {
+      document.removeEventListener('mouseup', handleClickOutside)
+    }
+  }, [handleClickOutside])
+
```

Comment:
This is already handled into the popup component it makes a duplicate. Maybe the issue is related to popup?